src/Illuminate/Support/ServiceProvider.php

```diff
@@ -132,13 +136,29 @@ protected function loadJsonTranslationsFrom($path)
      */
     protected function loadMigrationsFrom($paths)
     {
-        $this->app->afterResolving('migrator', function ($migrator) use ($paths) {
+        $this->callAfterResolving('migrator', function ($migrator) use ($paths) {
             foreach ((array) $paths as $path) {
                 $migrator->path($path);
             }
         });
     }
 
+    /**
+     * Setup an after resolving listener, or fire immediately if already resolved.
+     *
+     * @param  string  $name
+     * @param  callable  $callback
+     * @return void
+     */
+    protected function callAfterResolving($name, $callback)
+    {
+        $this->app->afterResolving($name, $callback);
+
+        if ($this->app->resolved($name)) {
+            $callback($this->app->make($name), $this->app);
+        }
+    }
```

Comment:
a bit curious why you add to listener first before checking if it's resolved.

Reply:
1. We always add the listener because there are two cases: (a) either it's not resolved yet, in which case, when it is resolved, then we'll run the callback, or (b) it is already resolved, but someone might clear the resolved instance from the container, and so next time its resolved, we still need to run the callback. So, in either case, we need to always register the callback.
2. The final part we need because if it is already resolved, we must run the callback immediately.

Reply:
I guess https://github.com/laravel/framework/blob/6.x/src/Illuminate/Support/Facades/Facade.php#L34-L42 need to follow the same structure too

Reply:
It already does? It doesn't matter which order the code is in.